Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -189,8 +192,28 @@ protected class ClassWriter : IClassWriter {
 
     private ConcreteSyntaxTree CreateFunction(string name, List<TypeArgumentInstantiation> typeArgs,
       List<Formal> formals, Type resultType, IToken tok, bool isStatic, bool createBody, MemberDecl member,
-      ConcreteSyntaxTree methodWriter, bool forBodyInheritance, bool lookasideBody) {
-      throw new NotImplementedException();
+      ConcreteSyntaxTree wr, bool forBodyInheritance, bool lookasideBody) {
+      if (!createBody) {
+        return null;
+      }
+
+      var customReceiver = !forBodyInheritance && NeedsCustomReceiver(member);
+      wr.Write("{0} {1}(", isStatic || customReceiver ? "def" : "", name);
```

Comment:
did you consider writing`"def "` and remove the space next to `{0}` ?

Reply:
I did, the else part here will be used for creating object of a class which will also need the space and. And once added in actual practice it will never print an empty string.